src/orion/core/io/config.py

```diff
@@ -0,0 +1,278 @@
+# -*- coding: utf-8 -*-
+"""
+:mod:`orion.core.io.config` -- Configuration object
+===================================================
+
+.. module:: config
+   :platform: Unix
+   :synopsis: Configuration object to define package configuration
+
+
+Highly inspired from https://github.com/mila-iqia/blocks/blob/master/blocks/config.py.
+
+"""
+import logging
+import os
+
+import yaml
+
+from orion.core.utils.flatten import flatten
+
+
+logger = logging.getLogger(__name__)
+
+
+NOT_SET = object()
+
+
+class ConfigurationError(Exception):
+    """Error raised when a configuration value is requested but not set."""
+
+
+class Configuration:
+    """Configuration object
+
+    Provides default values configurable at different levels. The configuration object can have
+    global default values, which may be overriden by user with yaml configuration files, environment
```

Comment:
overwritten*

Reply:
In this case I believe overridden (there was a typo) is more accurate, as we are not flushing the default value but rather adding another value which takes precedence, and removing it would revert the value of the config to the default value. (the config contains default value, yaml value, env var value and local value and provide the one with highest precedence).

Reply:
Makes sense.